app/api/dao/user_extension.py

```diff
@@ -95,22 +50,26 @@ def update_user_additional_info(data):
             return messages.USER_ID_IS_NOT_RETRIEVED, HTTPStatus.FORBIDDEN    
 
         timezone_value = data["timezone"]
-        timezone = Timezone(timezone_value).name
-        additional_info_data = {}
-        
-        existing_additional_info = UserExtensionModel.find_by_user_id(user_id)
-        if not existing_additional_info:
-            return messages.ADDITIONAL_INFORMATION_DOES_NOT_EXIST, HTTPStatus.NOT_FOUND
-        try:
-            existing_additional_info.is_organization_rep = data["is_organization_rep"]
-            existing_additional_info.timezone = timezone
-            additional_info_data["phone"] = data["phone"]
-            additional_info_data["mobile"] = data["mobile"]
-            additional_info_data["personal_website"] = data["personal_website"]
-        except KeyError as e:
-            return e, HTTPStatus.BAD_REQUEST
+        timezone = Timezone(timezone_value).name   
 
-        existing_additional_info.additional_info = additional_info_data
-        existing_additional_info.save_to_db()
-
-        return messages.ADDITIONAL_INFO_SUCCESSFULLY_UPDATED, HTTPStatus.OK
+        user_additional_info = UserExtensionModel.find_by_user_id(user_id)
+        if not user_additional_info:
+            user_additional_info = UserExtensionModel(user_id, timezone)
```

Comment:
Sorry I didn't know if userextension is synonymous to user_additional_info. If it is then this also seems fine 👍